server/serverPing.js

```diff
@@ -0,0 +1,15 @@
+const child_process = require("child_process"); //to run command-line tools
```

Comment:
Do you really need to run JavaScript for this? Or could you just run `ping $URL` on the command line directly?

Reply:
Hi Matthew, thank you for all your comments. Really appreciate it! I think the only dependencies that should be included are the ones from pingServer. For running ping $URL on the command line directly, I wasn't sure how to do it, so I will have a read and change it. If you have any documentation recommendation that would be great.

Reply:
Sure thing @munozirianni1988 - we're getting to the meat and potatoes of software engineering now!

I would look at `curl` for [making HTTP requests from the command line](https://medium.com/@dwyer.donovan/sending-http-requests-from-your-command-line-5ef6eecffb0a), e.g.:

```
curl https://google.com
```

To make sure the response is good, you could look at [HTTP status codes](https://superuser.com/questions/272265/getting-curl-to-output-http-status-code).

To see the requests that `curl` is making - or any app for that matter - you can use [HTTPdump](https://httpdump.app/). It allows you to create a fake website or API endpoint and see what your code is sending on requests.

I'd also recommend looking at the monthly usage limits for [Render](https://render.com/docs/free#monthly-usage-limits) to make sure that you can run the app all the time without running out of usage. You get 750 hours / month (31.25 days) but if those hours cover _both_ the server and the database, you would only get 15.125 days for those two instances.

If you can't run it 24 hours a day, I'd recommend running your cronjob only during the 12 hours you think it will be most useful (e.g. 8am - 8pm every day, or 14 hours a day but it's off on Sundays). You could add additional optimisation by [caching the database response](https://www.prisma.io/blog/benefits-and-challenges-of-caching-database-query-results-x2s9ei21e8kq#why-cache-database-query-results) in the API server, so we only hit the database once on startup (e.g. with [node-fetch-cache](https://www.npmjs.com/package/node-fetch-cache)). This would allow the database to be off most of the time, and the data only changes on deployment so it wouldn't get out-of-date.

Also make sure you look at the [Github Actions free tier limits](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) to make sure your cronjob will run all month. It may also help you ask questions like "Is it better to have a new job every 10 minutes or a 4-hour job that does something every 10 minutes?"

Reply:
Oh wow! Like we say in Cuba: el mango del arroz con mango! hahaha. I will dive into all this information as soon as I get home today. I wouldn't have a clue as to where to start looking/researching for all this, and all I was finding were very long complicated solutions. Thank you very much @mferryRV !